### DISCUSSION TITLE
Rockspec format version??? `rockspec_format`

### DISCUSSION THREAD

**Discussion by tmillr** [keplerproject/luarocks#1714] [Q&A] [answered]
I can't find any documentation on what this value (defined in a rockspec) accepts, nor what different `rockspec_format` versions mean/allow?

**Comment by alerque** [comment, marked as answer]
This is documented in the Wiki here on this project, but it's a little bit scattered. You'll have to find notes about specific fields mixed in with the rockspec docs.

The short version is that your main options are to use `rockspec_format = "1.0"` (or just leave it off) for widest compatibility or use `3.0` to get access to some LuaRocks v3 only features, but at the cost of not being usable by LuaRocks v2 and before. The extra things you can do are include developer only dependencies, specify tests, and a few things like that.

Personally my solution (see e.g. see my `fluent` or `decasify` rocks) is to use format 3 for the dev rockspec, then use a release transformation to strip the unsupported fields out of the rockspec when tagging releases to produce a rockspec with wide compatibility. This is useful for example to stay compatibly with Ubuntu LTS releases that still have aging tooling but allow developers access to nice stuff.

**Reply by tmillr** [reply]
Finally found this: https://github.com/luarocks/luarocks/blob/master/src/luarocks/type/rockspec.lua

**Reply by SnapperGee** [reply]
In rockspec version 3, there's a way to specify dev dependencies? I only see options for dependencies, test_dependencies, build_dependencies, and external_dependencies, but no option for developer dependencies. I would personally love and hope they add a dev dependencies section at some point, but do like a lot of the features and changes v3 brings!

**Comment by SnapperGee** [comment]
@alerque's answer answers the question, but I figured having a link to the official luarocks rockspec format documentation might be helpful:

https://github.com/luarocks/luarocks/blob/main/docs/rockspec_format.md

Not sure these docs existed at the time of this question, but I find the luarocks docs very helpful and complete.
